inference-engine/src/gna_plugin/transformations/swap_input_matmul_gna.hpp

```diff
@@ -2,15 +2,17 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
-#pragma once
+#ifndef SWAP_INPUT_MATMUL_GNA_HPP
+#define SWAP_INPUT_MATMUL_GNA_HPP
 
-#include <memory>
-#include <transformations_visibility.hpp>
+//#include <transformations_visibility.hpp>
 #include <ngraph/pass/graph_rewrite.hpp>
 
 namespace GNAPluginNS {
-
-// @brief Swaps and transposes inputs of MatMul if its first input is const and its batch size isn't supported by GNA
+// @brief Swaps and transposes inputs of MatMul if
+// 1. its first input is const and its batch size isn't supported by GNA
+// 2. its first input non-const and its batch size isn't supported by GNA
+// 3. its second input is const and its input batch size less than or equal to 8
```

Comment:
the 3-d case isn't quite clear for me, which input batch size should be less or equal to 8? and why is it required to swap inputs in this case?

Reply:
done